SocialCareCaseViewerApi/V1/UseCase/CaseStatusesUseCase.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using SocialCareCaseViewerApi.V1.Boundary.Response;
+using SocialCareCaseViewerApi.V1.Exceptions;
+using SocialCareCaseViewerApi.V1.Factories;
+using SocialCareCaseViewerApi.V1.Gateways;
+using SocialCareCaseViewerApi.V1.Domain;
+
+namespace SocialCareCaseViewerApi.V1.UseCase.Interfaces
+{
+    public class CaseStatusesUseCase : ICaseStatusesUseCase
+    {
+        private IDatabaseGateway _databaseGateway;
+
+        public CaseStatusesUseCase(IDatabaseGateway databaseGateway)
+        {
+            _databaseGateway = databaseGateway;
+        }
+
+        public ListCaseStatusesResponse ExecuteGet(long personId)
+        {
+            var caseStatus = _databaseGateway.GetCaseStatusesByPersonId(personId);
+
+            if (caseStatus == null || caseStatus.Count == 0)
+            {
+                throw new GetCaseStatusesException("Case status for person not found");
```

Comment:
Does this need to be an exception? If a person has no case statues isn't it fine to return an empty list?

Reply:
It returns an empty list!
Exception removed :)